src/setup-dotnet.ts

```diff
@@ -13,26 +13,28 @@ export async function run() {
     // If a valid version still can't be identified, nothing will be installed.
     // Proxy, auth, (etc) are still set up, even if no version is identified
     //
-    let version = core.getInput('dotnet-version');
-    if (!version) {
+    let versions = core.getMultilineInput('dotnet-version');
+    if (!versions.length) {
       // Try to fall back to global.json
       core.debug('No version found, trying to find version from global.json');
       const globalJsonPath = path.join(process.cwd(), 'global.json');
       if (fs.existsSync(globalJsonPath)) {
-        version = getVersionFromGlobalJson(globalJsonPath);
+        versions[0] = getVersionFromGlobalJson(globalJsonPath);
       }
     }
 
-    if (version) {
+    if (versions) {
       const includePrerelease: boolean =
         (core.getInput('include-prerelease') || 'false').toLowerCase() ===
         'true';
-
-      const dotnetInstaller = new installer.DotnetCoreInstaller(
-        version,
-        includePrerelease
-      );
-      await dotnetInstaller.installDotnet();
+      let dotnetInstaller: installer.DotnetCoreInstaller;
+      for (const version of versions) {
+        dotnetInstaller = new installer.DotnetCoreInstaller(
+          version,
+          includePrerelease
+        );
+        await dotnetInstaller.installDotnet();
```

Comment:
Could please add a check to DotnetCoreInstaller to add dotnet directory once. Current logic does not produce an error, but action can produce long PATH with repeating dotnet directory: `C:\Users\runneradmin\AppData\Local\Microsoft\dotnet;C:\Users\runneradmin\AppData\Local\Microsoft\dotnet;`

Reply:
I suggest move this code block:
```
if (process.env['DOTNET_INSTALL_DIR']) {
      core.addPath(process.env['DOTNET_INSTALL_DIR']);
      core.exportVariable('DOTNET_ROOT', process.env['DOTNET_INSTALL_DIR']);
    } else {
      if (IS_WINDOWS) {
        // This is the default set in install-dotnet.ps1
        core.addPath(
          path.join(process.env['LocalAppData'] + '', 'Microsoft', 'dotnet')
        );
        core.exportVariable(
          'DOTNET_ROOT',
          path.join(process.env['LocalAppData'] + '', 'Microsoft', 'dotnet')
        );
      } else {
        // This is the default set in install-dotnet.sh
        core.addPath(path.join(process.env['HOME'] + '', '.dotnet'));
        core.exportVariable(
          'DOTNET_ROOT',
          path.join(process.env['HOME'] + '', '.dotnet')
        );
      }
    }

    console.log(process.env['PATH']);
```
to the separated method, for example `addToPath` in `installer.ts` and invoke it once after iteration in `setup-dotnet.ts`

Reply:
addressed in latest commit